crates/ty/tests/cli.rs

```diff
@@ -1195,6 +1196,42 @@ fn concise_revealed_type() -> anyhow::Result<()> {
     Ok(())
 }
 
+#[test]
+fn can_handle_large_binop_expressions() -> anyhow::Result<()> {
+    let mut content = String::new();
+    writeln!(
+        &mut content,
+        "
+        from typing_extensions import reveal_type
+        total = 1{plus_one_repeated}
+        reveal_type(total)
+        ",
+        plus_one_repeated = " + 1".repeat(2000 - 1)
```

Comment:
I chose a *much* smaller limit than what we can handle (but still bigger than what we can handle on `main`) for two reasons:

* Debug builds take up more stack space and crash at smaller input sizes
* We don't want minor increases of stack frame sizes to lead to a test failure here.